docs/sharding.md

```diff
@@ -0,0 +1,107 @@
+  - [Usage Scenarios](#usage-scenarios)
+  - [Configuration method](#configuration-method)
+  - [Route algorithm](#route-algorithm)
+    - [Default behavior](#default-behavior)
+    - [Customization](#customization)
+    - [Implicitly pass HttpContext to endpoint router](#implicitly-pass-httpcontext-to-endpoint-router)
+
+<!-- /TOC -->
+
+## Usage Scenarios
+Routing logic is the way to decide to which SignalR Service instance among multiple instances your clients connect and your messages send. By applying different routing logic, this feature can be used in different scenarios. 
+* Scaling. Randomly route each client to one SignalR Service instance, send messages to all the SignalR Service instances so that you can scale the concurrent connections.
+* Cross-geo scenario. Cross-geo networks can be comparatively unstable. Route your clients to a SignalR Service instance in the same region can reduce cross-geo connections.
+* High availability and disaster recovery scenarios. Set up multiple service instances in different regions, so when one region is down, the others can be used as backup. Configure service instances as two roles, **primary** and **secondary**. By default, clients will be routed to a primary online instance. When SDK detects all the primary instances are down, it will route clients to secondary instances.
+
+## Configuration method
+
+To enable multiple SignalR Service instances, you should: 
+
+1. Use `Persistent` transport type.
+
+    Currently multiple-endpoint feature is only supported on `Persistent` mode. Add the following entry to your `local.settings.json` file locally or the application setting on Azure.
+
+    ```json
+    {
+        "AzureSignalRServiceTransportType":"Persistent"
+    }
+    ```
+
+2. Configure multiple SignalR Service endpoints entries in your configuration.
+
+    We use a [`ServiceEndpoint`](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs) object to represent a SignalR Service instance. You can define an service endpoint with its `<Name>` and `<EndpointType>` in the entry key, and the connection string in the entry value. The keys are in the following format : 
+
+    ```
+    Azure:SignalR:Endpoints:<Name>:<EndpointType>
+    ```
+
+    `<EndpointType>` is optional and is `primary` by default. See samples below:
+        
+    ```json
+    {
+        "Azure:SignalR:Endpoints:EastUs":"<ConnectionString>",
+        
+        "Azure:SignalR:Endpoints:EastUs2:Secondary":"<ConnectionString>",
+        
+        "Azure:SignalR:Endpoints:WestUs:Primary":"<ConnectionString>"
+    }
+    ```
+
+    > When you configure Azure SignalR endpoints in the App Service on Azure portal, don't forget to replace `":"` with `"__"`, the double underscore in the keys. For reasons, see [Environment variables](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0#environment-variables).
+
+## Route algorithm
```

Comment:
I think this section mix intro to default/custom router and step to make things work.
I prefer make a section inlcuding all steps to make things work, including config file, code change, etc; and another section to describe concepts.